main/views.py

```diff
@@ -2,9 +2,18 @@
 
 
 def home(request):
-    if request.user.is_authenticated:
+    if not request.user.is_authenticated:
```

Comment:
You should use the @login_required decorator instead of an if not authenticated

Reply:
I think that the `@login_required` redirects the user to the `home` view.
So it will create a recursive loop with no ending point.

Reply:
While I agree that we should use `@login_required`, I don't think that's the case here.
I want to display the both the landing page and the homepage under the `''` (base) URL.
If I use the decorator, then I wouldn't be able to do that since I would need to redirect to a page which has a different URL.
I mean I can use use the decorator, but it comes with a cost of weird URL patterns like `127.0.0.1/home` for the homepage or `127.0.0.1/welcome` for the welcome page.

Reply:
Exactly what Tamir said.

Reply:
I added the decorator for the `home` function.